src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndUpdateCommandResolver.java

```diff
@@ -35,7 +35,13 @@ public Class<FindOneAndUpdateCommand> getCommandClass() {
   public Operation resolveCommand(CommandContext ctx, FindOneAndUpdateCommand command) {
     ReadOperation readOperation = resolve(ctx, command);
     DocumentUpdater documentUpdater = DocumentUpdater.construct(command.updateClause());
-    return new ReadAndUpdateOperation(ctx, readOperation, documentUpdater, true, shredder);
+    boolean returnUpdatedDocument =
+        command.options() != null
+            && command.options().returnDocument() != null
```

Comment:
Can avoid null check by switching order of arguments:

   && "after".equals(command.options().returnDocument())

Reply:
Make sense. Changed the conditionality